draft-ietf-cose-hash-envelope.md

```diff
@@ -114,8 +111,7 @@ Hash_Envelope_as_COSE_Sign1 = [
 Hash_Envelope = #6.18(Hash_Envelope_as_COSE_Sign1)
 ~~~
 
-- Label `1` (alg)  Cryptographic algorithm to use
-- Label `16` (typ) MAY be used to assign a content format or media type to the entire hash envelope.
+- Label `1` (alg) Cryptographic algorithm to use
 - Label `TBD_1` (payload hash alg) MUST be present in the protected header and MUST NOT be present in the unprotected header.
 - Label `TBD_2` (content type of the preimage of the payload) MAY be present in the protected header or unprotected header.
 - Label `TBD_3` (payload_location) MAY be added to the protected header and MUST NOT be presented in the unprotected header.
```

Comment:
Not sure the payload location should be prohibited from the unprotected header.

I feel like we talked about this already... 

I think given these area all optional except for TBD_1, when present they should all be in the protected header, or they should all be allowed to be in either protected or unprotected, but not both.

Reply:
If a hint is unprotected, it also means someone could add it, redirecting the user to a hacked location. Protecting the location says the issuer states it is/was at that location, although the verifier may not have access to it. 

As the unprotected header is stripped, it would render it minimally useful, and don't think we should suggest it could be in the unprotected header

Reply:
You can't stop people from putting stuff in the unprotected header.
The proper security response to seeing some untrusted link, is not to click it.

By the same argument you give for location, I could trick a client that resolved the data to attempt to unzip it by setting the the content type to:

[application/vnd.cncf.helm.chart.content.v1.tar+gzip](https://www.iana.org/assignments/media-types/application/vnd.cncf.helm.chart.content.v1.tar+gzip)

Reply:
Is it better to not have a hint than an unprotected hint?

Reply:
I am not going to die on this hill: but...

something being optional does not imply that is not protection worthy, right? So there needs to be a better reason, I think.

Reply:
If we define a header parameter, we can say what it is intended to be used for.
If we don't want it to be in the unprotected header, we can say so.
This should encourage implementations to not accept it there.
(This can be extended to a "MUST be ignored" in the unprotected header.)

Reply:
Is there any such apatite for such strong normative language? I don't think that we want to be restrictive in such a manner here. Any counter indications?

Reply:
> given that content type and location are already optional, we admit that it is ok for them to be unprotected.
Being optional doesn't preclude being protected, so I admin nothing :)

I'd vote for leaving the text as-is, por favor

Reply:
Resolving based on the above discussion. Please activate/block if you feel differently.